Documentation for blueprint product packaging (#133)
eclipse-theia/theia-website: src/docs/blueprint_download.md

---

> title: Eclipse Theia Blueprint Product

Reviewer: ```suggestion
title: Eclipse Theia Blueprint
```

---

> # Eclipse Theia Blueprint Product

Reviewer: ```suggestion
# Eclipse Theia Blueprint
```

<img src="/theia-logo-blueprint.png" alt="Theia Logo Blue" style="max-width: 525px">

> ## What is this?

Reviewer: ```suggestion
## What is Eclipse Theia Blueprint?
```

> Theia Blueprint product is a **template** to showcase the capabilities of Theia as well as how to build desktop-based products based on the platform. Theia Blueprint assembles a selected subset of existing Theia features and extensions. We provide installers for Theia Blueprint to be downloaded (see below) as well as documentation on how to customize this template to build a product and installers for your own Theia-based product.

Reviewer: ```suggestion
Eclipse Theia Blueprint is a **template** for building desktop-based products based on the Eclipse Theia platform, as well as to showcase Eclipse Theia capabilities. TIt is made up of a  subset of existing Eclipse Theia features and extensions and can be easily downloaded and installed on all major operating system platforms (see below). Documentation is available to help you customize and build your own Eclipse Theia-based product.
```

## What is it not?

> Theia Blueprint product is ***not*** **a production-ready product**. Therefore, it is also not a replacement for Visual Studio Code or any other IDE.

Reviewer: ```suggestion
Eclipse Theia Blueprint is ***not*** **a production-ready product**. Therefore, it is also not meant to be a replacement for Visual Studio Code or any other IDE.
```

## Downloads

<span style="color:red">Warning: Currently Theia Blueprint product is an **alpha** version. Please consider the current versions as experimental. </span>

<table cellspacing="25">
  <thead>
    <tr>
      <th>Windows</th>
      <th>Linux</th>

>       <th>MacOS</th>

Reviewer: ```suggestion
      <th>macOS</th>
```

    </tr>
  </thead>
  <tbody>
    <tr>
      <td><a href="https://www.eclipse.org/downloads/download.php?file=/theia/1.9.1/windows/Theia-1.9.1.exe&r=1">Download 1.9.1</a></td>
      <td><a href="https://www.eclipse.org/downloads/download.php?file=/theia/1.9.1/linux/Theia-1.9.1.AppImage&r=1">Download 1.9.1</a></td>

>       <td><a href="https://www.eclipse.org/downloads/download.php?file=/theia/1.9.1/macos/Theia-1.9.1.pkg&r=1">Download 1.9.1</a></td>

Reviewer 2: Can we not instead point to the `latest` link, this would mean we would not need to maintain the documentation on each update. 

For example `1.9.1` is actually outdated before even being merged:

- https://download.eclipse.org/theia/latest/linux/
- https://download.eclipse.org/theia/latest/macos/
- https://download.eclipse.org/theia/latest/windows/

I think pointing to the actual download location rather than executing the download itself would be more appropriate.

Author: I think with this approach we will lose download stats, as we have to use the "select a mirror script" for a file (https://wiki.eclipse.org/Project_Download_Stats#ZIP_files)
But I will check if we can publish the files in a way, so we can have direct links to the latest version

Author: Fixed with https://github.com/eclipse-theia/theia-blueprint/issues/51

>       <td><a href="https://www.eclipse.org/downloads/download.php?file=/theia/1.9.0/macos/Theia-1.9.0.pkg&r=1">Download 1.9.0</a></td>

Reviewer 2: I believe we should reference a constant url to download, referencing a version will become outdated pretty quickly and will require too much maintenance on the website.

Author: While we deploy all the latest versions to https://download.eclipse.org/theia/latest/ as well, the file names still contain the version number. So at least for now we don't have a static link to the last version. We will have to adapt the deployment scripts first.

    </tr>
  </tbody>
</table>

> ## Reporting feature requests and bugs

Reviewer: ```suggestion
## Reporting Feature Requests and Bugs
```

> The features in the Theia Blueprint product are based on Theia and the included extensions/plugins. For more details please see the [Eclipse Theia Github Repository](https://github.com/eclipse-theia/theia). 

Reviewer: ```suggestion
The features found in Eclipse Theia Blueprint are based on Eclipse Theia and the available extensions/plugins. For more details please see the [Eclipse Theia Github Repository](https://github.com/eclipse-theia/theia). 
```

> The features in the Eclipse Theia Blueprint product are based on Theia and the included extensions/plugins. For bugs in Theia please consider opening an issue in the [Theia project on Github](https://github.com/eclipse-theia/theia/issues/new/choose). 

Reviewer 2: I believe the goal is not to have users report `blueprint` problems directly in the upstream repository, but `blueprint` itself (given that it is also not a production-ready product). Can you confirm @marcdumais-work.

Reviewer 3: I think `blueprint`-specific issues should be opened in that repo. But here the mention is for `bugs in Theia`, which indeed are better opened in the main Theia repo. I am not sure how useful it is to mention issues here. Maybe it could be rephrased more generally, still pointing to the Theia repo? e.g. "For more details please see the [Eclipse Theia  repo](https://github.com/eclipse-theia/theia/)"

> Theia Blueprint product only packages existing functionality into a product and installers for the product. If you believe there is a mistake in packaging, something needs to be added to the packaging or the installers do not work properly, please [open an issue on Github](https://github.com/eclipse-theia/theia-blueprint/issues/new/choose) to let us know.

Reviewer: ```suggestion
Eclipse Theia Blueprint only packages existing Eclipse Theia functionality and installers. If you believe there is a mistake in packaging, something needs to be added to the packaging or the installers do not work properly, please [open an issue on Github](https://github.com/eclipse-theia/theia-blueprint/issues/new/choose) to let us know.
```

## Source Code

> The Theia Blueprint source code is available on [Github](https://github.com/eclipse-theia/theia-blueprint).

Reviewer: ```suggestion
The source code for Eclipse Theia Blueprint is available on [Github](https://github.com/eclipse-theia/theia-blueprint).
```

> The source code of the Theia Blueprint product is available on [Github](https://github.com/eclipse-theia/theia-example).

Reviewer 2: ```suggestion
The Theia Blueprint source code is available on [Github](https://github.com/eclipse-theia/theia-example).
```

## Documentation

> Please see [here](../blueprint_documentation) for documentation on how to customize the Theia Blueprint product.

Reviewer: ```suggestion
Please see [here](../blueprint_documentation) for documentation on how to customize Eclipse Theia Blueprint.
```

> Please see [here](../blueprint_documentation) for documentation how to customize the Eclipse Theia Blueprint product.

Reviewer 2: ```suggestion
Please see [here](../blueprint_documentation) for documentation on how to customize the Eclipse Theia Blueprint product.
```
